compiler/optimizer/LocalOpts.cpp

```diff
@@ -6637,9 +6637,17 @@ TR::Block *TR_BlockSplitter::splitBlock(TR::Block *pred, TR_LinkHeadAndTail<Bloc
             {
             if (lastRealNode->getBranchDestination()->getNode()->getBlock()->getNumber() == candidate->getNumber())
                {
-               lastRealNode->reverseBranch(cloneEnd->getExit()->getNextTreeTop());
+               TR::TreeTop *newBranchDestinationTT = cloneEnd->getExit()->getNextTreeTop();
+
+               lastRealNode->reverseBranch(newBranchDestinationTT);
+               if (!pred->hasSuccessor(newBranchDestinationTT->getNode()->getBlock()))
```

Comment:
Should there be a short comment 'in the code' for why this conditional code is needed ? Since it is for a specific edge case, maybe this can be mentioned to make it easy for someone reading the code without having to do the archeology to get to this PR and the discussion therein.

Reply:
Addressed in [6cb52ac](https://github.com/eclipse-omr/omr/compare/f30e6834646f6840ec030740a3b3987381aad2c9..6cb52ac61bfc71965274d752a3372ffbc66fd1c6). Ready for another review. Thanks!